core/src/avm2/script.rs

```diff
@@ -129,11 +130,25 @@ impl<'gc> TranslationUnit<'gc> {
             return Ok(method.clone());
         }
 
+        let is_global = read.domain.is_avm2_global_domain(activation);
         drop(read);
 
-        let method: Result<Gc<'gc, BytecodeMethod<'gc>>, Error> =
-            BytecodeMethod::from_method_index(self, method_index, is_function, activation);
-        let method: Method<'gc> = method?.into();
+        let bc_method: Gc<'gc, BytecodeMethod<'gc>> =
+            BytecodeMethod::from_method_index(self, method_index, is_function, activation)?;
+
+        let mut method = None;
+        if is_global {
+            if let Some(native) = activation.avm2().native_table[method_index.0 as usize] {
```

Comment:
Maybe this can be included in the generated `NATIVE_TABLE` itself? Though it would require `Method` to be constructible at a `const` context.

Reply:
I don't think this is nearly as easy, with GC in the way.
One minor thing I think we could do would be move the props (esp `signature`) from `BytecodeMethod` to the `NativeMethod` without cloning them.

Reply:
I've refactored this to move out of `signature`.